src/storage/query/QueryBaseProcessor.inl

```diff
@@ -163,10 +165,85 @@ cpp2::ErrorCode QueryBaseProcessor<REQ, RESP>::checkAndBuildContexts(const REQ&
         exp_->setContext(expCtx_.get());
     }
 
-    buildRespSchema();
+    buildTTLInfoAndRespSchema();
     return cpp2::ErrorCode::SUCCEEDED;
 }
 
+template<typename REQ, typename RESP>
+bool QueryBaseProcessor<REQ, RESP>::tagHasTTL(TagID tagId) {
+    auto tagFound = tagTTLInfo_.find(tagId);
+    if (tagFound != tagTTLInfo_.end()) {
+        return true;
+    }
+    return false;
+}
+
+template<typename REQ, typename RESP>
+bool QueryBaseProcessor<REQ, RESP>::edgeHasTTL(EdgeType edgeType) {
+    auto edgeFound = edgeTTLInfo_.find(edgeType);
+    if (edgeFound != edgeTTLInfo_.end()) {
+        return true;
+    }
+    return false;
+}
+
+template<typename REQ, typename RESP>
+std::pair<std::string, int64_t>  QueryBaseProcessor<REQ, RESP>::getTagTTLInfo(TagID tagId) {
+    auto tagFound = tagTTLInfo_.find(tagId);
+    if (tagFound != tagTTLInfo_.end()) {
+        return tagFound->second;
+    }
+    return std::make_pair(std::string(""), 0);
+}
+
+template<typename REQ, typename RESP>
+std::pair<std::string, int64_t>  QueryBaseProcessor<REQ, RESP>::getEdgeTTLInfo(EdgeType edgeType) {
+    auto edgeFound = edgeTTLInfo_.find(edgeType);
+    if (edgeFound != edgeTTLInfo_.end()) {
+        return edgeFound->second;
+    }
+    return std::make_pair(std::string(""), 0);
+}
+
+template<typename REQ, typename RESP>
+bool QueryBaseProcessor<REQ, RESP>::checkDataExpiredForTTL(const meta::SchemaProviderIf* schema,
```

Comment:
Could you unify the methods with checkDataTtlValid inside compacionFilter

Reply:
fix